torchgeo/datasets/geo.py

```diff
@@ -603,20 +598,33 @@ def _cached_load_warp_file(self, filepath: Path) -> DatasetReader:
         """
         return self._load_warp_file(filepath)
 
-    def _load_warp_file(self, filepath: Path) -> DatasetReader:
+    def _load_warp_file(self, filepath: Path, crs: CRS | None = None) -> DatasetReader:
         """Load and warp a file to the correct CRS and resolution.
 
         Args:
             filepath: file to load and warp
+            crs: Optionally specify which CRS to reproject to. This is used in __init__
+                as self.index.crs is not defined at this point.
 
         Returns:
             file handle of warped VRT
         """
+        if crs is None and hasattr(self, 'index'):
+            crs = self.crs
+
         src = rasterio.open(filepath)
 
+        # See if file has a color map
+        if len(self.cmap) == 0:
+            try:
+                self.cmap = src.colormap(1)  # type: ignore[misc]
```

Comment:
Not sure what how/when this is used. I moved it from `__init__` as I'm not sure if colormap is preserved in the vrt. If it is, I think it is better to move it back. Or into the special dataset implementations that use it